sdks/java/extensions/sorter/src/test/java/org/apache/beam/sdk/extensions/sorter/BufferedExternalSorterTest.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.extensions.sorter;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Constructor;
+import java.util.Arrays;
+import org.apache.beam.sdk.extensions.sorter.SorterTestUtils.SorterGenerator;
+import org.apache.beam.sdk.values.KV;
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link BufferedExternalSorter}. */
+public class BufferedExternalSorterTest {
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testNoFallback() throws Exception {
+    ExternalSorter mockExternalSorter = mock(ExternalSorter.class);
+    InMemorySorter mockInMemorySorter = mock(InMemorySorter.class);
+    Constructor<BufferedExternalSorter> constructor =
+        BufferedExternalSorter.class
+            .getDeclaredConstructor(ExternalSorter.class, InMemorySorter.class);
+    constructor.setAccessible(true);
+    BufferedExternalSorter testSorter =
```

Comment:
Changed. Have worked in previous projects where the standard was to use reflection rather than change accessibility to package-private for the sake of testing.